Add links for Debug UI to GraphiQL

a-limyr opened the pull request:
### Summary

This feature provides the following new functionality for the debug client:
A new button that when pressed will take the parameters set and create a GraphQL trip query that opens in GraphiQL.
Links in the leg lists will now take you to GraphQL queries in GraphiQL for the different entities.

### Issue
This should resolve issue 1. and 2. listen under this: https://github.com/opentripplanner/OpenTripPlanner/issues/5330#issuecomment-2273311110 issue.

testower commented on client/src/components/ItineraryList/ItineraryLegDetails.tsx:
```diff
@@ -17,19 +29,41 @@ export function ItineraryLegDetails({ leg, isLast }: { leg: Leg; isLast: boolean
         {leg.line && (
           <>
-            <u>
+            <a
```
I would consider a new component here similar to GraphiQLRouteButton to make this a little bit easier on the eyes.
Something along these lines:
```typescript
<GraphiQLLink title={leg.line?.id} href={...}>
  {leg.line.publicCode} {leg.toEstimatedCall?.destinationDisplay?.frontText}
</GraphiQLLink>
```

leonardehrenfried replied:
I would go even further and build the URL in the component.

testower commented on client/src/components/SearchBar/GraphiQLRouteButton.tsx:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <div className="search-bar-route-button-wrapper">
+      <Button href={graphiQLUrl + '&query=' + formattedQuery + '&variables=' + formattedVariables} target={'_blank'}>
+        GraphQL
```
```suggestion
        GraphiQL
```

a-limyr commented:
I have refactored and created the following elements:

- ItineraryGraphiQLLineLink
- ItineraryGraphiQLQuayLink
- ItineraryGraphiQLAuthorityLink

I have also added the functionality to click on authority to create a graphQL query.

leonardehrenfried commented:
I will let @testower do the proper review but I was thinking of a very generic component like:

```
<GraphiQLLink query={query} args={args} text={leg.name}/>
```

testower commented:
What @leonardehrenfried said is also what I had in mind. The text can also just be the child(ren) of the component like i suggested initially.

testower commented:
Actually, what Leonard suggested is even more sophisticated.

a-limyr commented:
Ok, the idea behind creating different link types where to remove the need for the encapsulating element from fetching the different query types (line, quay etc.), but maybe adding type as an argument and make the link element handle the query logic (matching type with query).  This will result in something like this:

<GraphiQLLink args={args} text={leg.name}/>

Would that be ok?

leonardehrenfried commented:
How would that component figure out the correct query to use?

a-limyr commented:
With use of a QueryType variable, it can either be part of args:

`<GraphiQLLink args={args} text={leg.name}/>`

or as a separat argument:

`<GraphiQLLink args={args} type={type} text={leg.name}/>`

Any preference for an approach?

leonardehrenfried commented:
Not really. @testower might have an opinion.

testower commented:
No strong preference. To be perfectly honest, I think the current solution works well enough that we don't have to spend more time on it now. Maybe in the future if we add more graphiql-links we could revisit this topic.

leonardehrenfried reviewed (approved):
If it's good enough for testower, it's good enough for me.
